protocol/avdl/keybase1/upk.avdl

```diff
@@ -0,0 +1,67 @@
+@namespace("keybase.1")
+protocol UPK {
+
+  enum KeyType {
+      NONE_0,
+      NACL_1,
+      PGP_2
+  }
+
+  record MerkleRootV2 {
+      Seqno seqno;
+      bytes hashMeta;
+  }
+
+  record SignatureTime {
+      MerkleRootV2 merkleRootAtSig;
+      union { null, MerkleRootV2 } firstAppearedIn;
+      Time time;
+  }
+
+  record PublicKeyV2Base {
+      KID kid;
+      boolean isSibkey;
+      boolean isEldest;
+      Time cTime;
+      Time eTime;
+      SignatureTime provisioning;
+      union { null, SignatureTime } revocation;
+  }
+
+  record PublicKeyV2NaCl {
+      PublicKeyV2Base base;
+      union { null, KID } parent;
+      DeviceID deviceID;
+      string deviceDescription;
+      string deviceType;
+  }
+
+  fixed PGPFingerprint(20);
+
+  record PublicKeyV2PGPSummary {
+      PublicKeyV2Base base;
+      PGPFingerprint fingerprint;
+      array<PGPIdentity> identities;
+  }
+
+  variant PublicKeyV2 switch (KeyType keyType) {
+      case NACL: PublicKeyV2NaCl;
+      case PGP : PublicKeyV2PGPSummary;
+      default  : void;
+  }
+
+  record UserPlusKeysV2 {
+      UID uid;
+      string username;
+      Seqno eldestSeqno;
+      UserVersionVector uvv;
+      array<PerUserKey> perUserKeys;
```

Comment:
You know I thought about that a bit, but yes, to sort them all generation ascending. Otherwise we'll have to duplicate the primary key inside and outside the object.